Userland/Libraries/LibGfx/VectorN.h

```diff
@@ -139,6 +140,18 @@ requires(N >= 2 && N <= 4) class VectorN final {
         return result;
     }
 
+    [[nodiscard]] constexpr VectorN operator*(Matrix<N, T> const& matrix) const
+    {
+        VectorN result;
+        for (auto i = 0u; i < N; ++i) {
+            result[i] = 0;
+            for (auto j = 0u; j < N; ++j) {
+                result[i] += matrix.elements()[j][i] * data()[j];
+            }
+        }
+        return result;
+    }
```

Comment:
Currently we define some unrolled vector transformations in `LibGfx/Matrix(..).h`, I think it's better to have the generalized version of this in the same style in `Matrix.h` instead so we have them close together.

Reply:
You are right I overlooked `Matrix4x4.h:20`, I just used that so the new Matrix * Vector function was not needed in the end